### PR TITLE
feat: better ButtonGroup docs and fixed spaced implementation

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

- Better documentation for the `ButtonGroup` component with more real-world examples.
- Fixes wrong `spaced` implementation (there was 2 * spacing space between items) by using `Stack` component in the implementation.
- Fixes broken `withDivider` implementation

### CODE REVIEW

File: packages/components/button/examples/ButtonGroupCollapsedExample.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  Stack,
+  ButtonGroup,
+  Button,
+  IconButton,
+} from '@contentful/f36-components';
+import { ChevronDownIcon, PlusIcon } from '@contentful/f36-icons';
+
+export default function ButtonGroupCollapsedExample() {
+  return (
+    <Stack>
```
Comment (gui-santos) [inline packages/components/button/examples/ButtonGroupCollapsedExample.tsx:12]:
```suggestion
    <Stack flexDirection="column">
```

Since this two groups are different examples (I'm assuming this because of the text before them)
I would split them or show them in a column otherwise it feels to me that they are part of the same "thing"

**Review by gui-santos — APPROVED** [review]
Only a small suggestion

File: packages/components/button/src/ButtonGroup/README.mdx [hunk 2]
```diff
@@ -5,61 +5,44 @@ status: 'stable'
...
-    variant="positive"
-    aria-label="Open dropdown"
-    startIcon={<ChevronDownIcon />}
-  />
-</ButtonGroup>
-```
+### Collapsed
 
-It's possible to add a divider for buttons that don't have border:
+The most common example is to group primary button with a secondary button or with a button that opens a dropdown with secondary actions.
```
Comment (burakukula) [inline packages/components/button/src/ButtonGroup/README.mdx:23]: hmm, we say here that it is a common case to group primary and secondary button, but we don't show it in the example. Is it really that common btw?

File: packages/components/button/src/ButtonGroup/README.mdx [hunk 3]
```diff
@@ -5,61 +5,44 @@ status: 'stable'
...
-  <Button variant="secondary">Button</Button>
-  <Button variant="secondary">Button</Button>
-  <Button
-    variant="positive"
-    aria-label="Open dropdown"
-    startIcon={<ChevronDownIcon />}
-  />
-</ButtonGroup>
-```
+### Collapsed
```
Comment (burakukula) [inline packages/components/button/src/ButtonGroup/README.mdx:21]: Why we call it collapsed? 🤔

File: packages/components/button/src/ButtonGroup/README.mdx [hunk 4]
```diff
@@ -5,61 +5,44 @@ status: 'stable'
...
-    Button
-  </Button>
-  <Button variant="positive" startIcon={<ChevronDownIcon />} />
-</ButtonGroup>
 ```
 
-Example of Group Button with spaced variant
+### Spaced
+
+Instead of collapsing buttons all together you can use `variant="spaced"` to make them spaced and control an empty space with `spacing` property.
```
Comment (burakukula) [inline packages/components/button/src/ButtonGroup/README.mdx:33]: hmm, I don't get why we say collapsing buttons. but maybe it is my English hehehe. I always thought collapsing might be relevant to something like an accordion, menu, something that opens and closes.

Comment (burakukula) [reply packages/components/button/src/ButtonGroup/README.mdx:33]: We could say just grouped (without space, if we want to be very specific)? dunno. maybe it's just me.

File: packages/components/button/src/ButtonGroup/README.mdx [hunk 5]
```diff
@@ -5,61 +5,44 @@ status: 'stable'
...
-```
+### Collapsed
 
-It's possible to add a divider for buttons that don't have border:
+The most common example is to group primary button with a secondary button or with a button that opens a dropdown with secondary actions.
 
-```jsx
-// import { ButtonGroup, Button } from '@contentful/f36-button';
-// import { ChevronDownIcon, PlusIcon } from '@contentful/f36-icons';
+Also, it is possible to add a divider for buttons that don't have border.
```
Comment (burakukula) [inline packages/components/button/src/ButtonGroup/README.mdx:25]: What do you mean by that? which buttons don’t have borders?

File: packages/components/button/examples/ButtonGroupCollapsedExample.tsx (hunk 1 above)
Comment (suevalov) [reply packages/components/button/examples/ButtonGroupCollapsedExample.tsx:12]: 👍🏼

File: packages/components/button/src/ButtonGroup/README.mdx (hunk 2 above)
Comment (suevalov) [reply packages/components/button/src/ButtonGroup/README.mdx:23]: I'm going to rephrase it to `One of the common use cases...` and we actually show this in the example below, aren't we?

File: packages/components/button/src/ButtonGroup/README.mdx (hunk 3 above)
Comment (suevalov) [reply packages/components/button/src/ButtonGroup/README.mdx:21]: Because this variant is called `collapsed` 🤔

File: packages/components/button/src/ButtonGroup/README.mdx (hunk 4 above)
Comment (suevalov) [reply packages/components/button/src/ButtonGroup/README.mdx:33]: Hmm... I'm questioning it now too, but I guess it's too late to change the name before the release. The better name would be `merged` or something like this. WDYT?

File: packages/components/button/src/ButtonGroup/README.mdx (hunk 5 above)
Comment (suevalov) [reply packages/components/button/src/ButtonGroup/README.mdx]: updated with an explanation

File: packages/components/button/src/ButtonGroup/README.mdx (hunk 4 above)
Comment (burakukula) [reply packages/components/button/src/ButtonGroup/README.mdx:33]: yeah, that sounds better to me. Collapsed... hmmm. why we called the prop collapsed? 😄

File: packages/components/button/src/ButtonGroup/README.mdx [hunk 6]
```diff
@@ -5,61 +5,44 @@ status: 'stable'
...
 slug: /components/button-group/
 github: 'https://github.com/contentful/forma-36/tree/forma-v4/packages/components/button/src/ButtonGroup'
-storybook: 'https://v4-f36-storybook.netlify.app/?path=/story/components-button-buttongroup--basic'
 typescript: ./ButtonGroup.tsx
 ---
 
-import { Props } from '@contentful/f36-docs-utils';
-import { Button } from '@contentful/f36-button';
-import { ChevronDownIcon, PlusIcon } from '@contentful/f36-icons';
+`ButtonGroup` should be used to group buttons whose actions are related, with an option to collapse them together or split them with an equal amount of free space.
```
Comment (burakukula) [inline packages/components/button/src/ButtonGroup/README.mdx:11]:
```suggestion
`ButtonGroup` should be used to group buttons whose actions are related, with an option to merge them together or split them with an equal amount of free space.
```

**Review by burakukula — APPROVED** [review]

File: packages/components/button/src/ButtonGroup/README.mdx (hunk 4 above)
Comment (suevalov) [reply packages/components/button/src/ButtonGroup/README.mdx:33]: I have no idea 😢 but I think it's too late to make a breaking change. Should we create a new variant `merged` and keep `collapsed` as a synonym of `merged` for backward compatibility? 

We can remove `collapsed` in v5.

**Review by thebesson — APPROVED** [review]
